include/ginkgo/core/base/dim.hpp

```diff
@@ -200,11 +215,154 @@ struct dim<1u, DimensionType> {
         return dim(x.first_ * y.first_);
     }
 
+    friend GKO_ATTRIBUTES std::ostream &operator<<(std::ostream &os,
+                                                   const dim &x)
+    {
+        os << x.first_;
+        return os;
+    }
+
 private:
     dimension_type first_;
 };
 
 
+/**
+ * A type representing the dimensions of a multidimensional batch object.
+ *
+ * @tparam Dimensionality  number of dimensions of the object
+ * @tparam DimensionType  datatype used to represent each dimension
+ *
+ * @ingroup batch_dim
+ */
+template <size_type Dimensionality = 2, typename DimensionType = size_type>
+struct batch_dim {
+    static constexpr size_type dimensionality = Dimensionality;
+    using dimension_type = DimensionType;
+
+    /**
+     * Checks if the batch_dim object stores equal sizes.
+     *
+     * @return bool representing whether equal sizes are being stored
+     */
+    bool stores_equal_sizes() const { return equal_sizes_; }
+
+    /**
+     * Get the number of batches stored
+     *
+     * @return num_batches
+     */
+    size_type get_num_batches() const { return num_batches_; }
```

Comment:
IMO, "num_batches" is counter intuitive and makes one think of many batches. To have consistent English semantics, we should call this the number of batch-entries, but I felt "batch size" would be better suited for code. That is, a batch is made of `n` matrices, so the size of the batch is `n`.
```suggestion
    size_type get_batch_size() const { return num_batches_; }
```

Reply:
I guess `get_num_batch_entries` is better, but I feel that `batch_size` is confusing. I will change it to `get_num_batch_entries`